core/Pos/Crypto/HD.hs

```diff
@@ -77,31 +84,78 @@ deriveHDSecretKey passPhrase (SecretKey xprv) childIndex
   | otherwise =
       SecretKey $ deriveXPrv passPhrase xprv (childIndex - maxHardened - 1)
 
+addrAttrNonce :: ByteString
+addrAttrNonce = "serokellfore"
+
 -- | Serialize tree path and encrypt it using passphrase via ChaChaPoly1305.
 packHDAddressAttr :: HDPassphrase -> [Word32] -> HDAddressPayload
-packHDAddressAttr (HDPassphrase simkey) path = do
-    let pathSer = encodeStrict path
-    let packCF =
+packHDAddressAttr (HDPassphrase passphrase) path = do
+    let !pathSer = encodeStrict path
+    let !packCF =
           encryptChaChaPoly
-              "serokellfore"
-              simkey
-              "addressattr" -- I don't know what does mean this parameter.
-                            -- Seems value isn't important...
+              addrAttrNonce
+              passphrase
+              ""
               pathSer
     case packCF of
-        CryptoFailed er -> error $ "Error during packHDAddressAttr: " <> show er
+        CryptoFailed er -> error $ "Error in packHDAddressAttr: " <> show er
         CryptoPassed p  -> HDAddressPayload p
 
+unpackHDAddressAttr :: MonadFail m => HDPassphrase -> HDAddressPayload -> m [Word32]
```

Comment:
Opportunity to generate `HDAddressPayload` within CSL-879, with which nothing can be done then, makes me :smile: